stable/distribution/templates/_helpers.tpl

```diff
@@ -79,3 +79,83 @@ Scheme (http/https) based on Access TLS enabled/disabled
 {{- printf "%s" "http" -}}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Resolve joinKey value
+*/}}
+{{- define "distribution.joinKey" -}}
+{{- default .Values.distribution.joinKey .Values.global.joinKey -}}
+{{- end -}}
+
+{{/*
+Resolve masterKey value
+*/}}
+{{- define "distribution.masterKey" -}}
+{{- default .Values.distribution.masterKey .Values.global.masterKey -}}
+{{- end -}}
+
+{{/*
+Resolve joinKeySecretName value
+*/}}
+{{- define "distribution.joinKeySecretName" -}}
+{{- default (default (include "distribution.fullname" .) .Values.distribution.joinKeySecretName) .Values.global.joinKeySecretName -}}
+{{- end -}}
+
+{{/*
+Resolve masterKeySecretName value
+*/}}
+{{- define "distribution.masterKeySecretName" -}}
+{{- default (default (include "distribution.fullname" .) .Values.distribution.masterKeySecretName) .Values.global.masterKeySecretName -}}
+{{- end -}}
+
+{{/*
+Resolve imagePullSecrets value
+*/}}
+{{- define "distribution.imagePullSecrets" -}}
+{{- if .Values.global.imagePullSecrets }}
+imagePullSecrets:
+{{- range .Values.global.imagePullSecrets }}
+  - name: {{ . }}
+{{- end }}
+{{- else if .Values.imagePullSecrets }}
```

Comment:
Shouldn’t the non-global value be first? I mean, doesn’t the specific values take precedence over the global ones?

Reply:
No.  Global value from parent chart takes precedence over subchart value